scripts/nbconverted/main_Pa_sim_enhance_AtoB.py

```diff
@@ -0,0 +1,225 @@
+
+# Size of the gene set that will be regulated by gene A
+geneSetB_size = 1000 
+
+# Percentage to upregulate each gene in set B
+effect_size = 0.5
+
+# Threshold for activation of gene A 
+thresholdA = 0.5
+
+# Name of analysis
+analysis_name = 'sim_AB_2775_300_skewA'
+
+
+# In[3]:
+
+
+# Load arguments
+data_file = os.path.join(os.path.dirname(os.getcwd()), "data", "all-pseudomonas-gene-normalized.zip")
+
+
+# In[4]:
+
+
+# Read in data
+data = pd.read_table(data_file, header=0, sep='\t', index_col=0, compression='zip').T
+data.head(5)
+
+
+# In[5]:
+
+
+# Randomly select genes for set A
+gene_ids = list(data.columns)
+geneSetA = random.sample(gene_ids, geneSetA_size)
+
+
+# In[6]:
+
+
+# checkpoint:  Check the number of genes
+assert(len(gene_ids)==5549)
+print("Confirmed that all gene ids are available")
+
+
+# In[7]:
+
+
+# Remove gene ids that were selected for gene set A
+gene_ids = [elem for elem in gene_ids if elem not in geneSetA]
+print("After removing {} gene ids for set A, there are {} gene ids remaining.".format(geneSetA_size, len(gene_ids)))
+
+
+# In[8]:
+
+
+# Randomly select genes for gene set B
+geneSetB = random.sample(gene_ids, geneSetB_size)
+
+# Remove gene ids that were selected for gene set B
+gene_ids = [elem for elem in gene_ids if elem not in geneSetB]
+print("After removing {} gene ids for set B, there are {} gene ids remaining.".format(geneSetB_size, len(gene_ids)))
+
+
+# In[9]:
+
+
+# checkpoint:  Check that genes in set A are distinct from genes in set B
+assert(len(set(geneSetA).intersection(set(geneSetB))) == 0)
+
+
+# In[10]:
+
+
+# Main input data simulation 
+
+# Loop through all samples
+num_samples = data.shape[1]
+for sample_id in data.index:
+    row = data.loc[sample_id]
+    
+    # Randomly select a value [0,1] where each value is uniformly likely to be chosen
+    new_A_exp = random.uniform(0.4, 1.0)
+    
+    # Set gene expression value for genes in set A to be the same random value selected
+    data.loc[sample_id][geneSetA] = new_A_exp
+    
+    # Select sample gene from set A to be representative since the expression is the same
+    # for all genes in the set
+    geneSetA_pick = geneSetA[0]
+    
+    # Check if expression of genes in set A exceed the threshold
+    if data.loc[sample_id,geneSetA_pick] > thresholdA:
+        
+        # Scale genes by some fixed percentage
+        for gene in geneSetB:
+            data.loc[sample_id,gene] = (1+effect_size)*data.loc[sample_id,gene]            
+            
+# if any exceed 1 then set to 1 since gene expression is normalized
+data[data>=1.0] = 1.0
+
+
+# In[11]:
+
+
+# Dataframe with only genes in set A
+geneA_only = pd.DataFrame(data[geneSetA_pick], index=data.index, columns=[geneSetA_pick])
+
+# Drop genes in set A
+data_holdout = data.drop(geneSetA, axis=1)
+
+geneA_only
+
+
+# In[12]:
+
+
+# checkpoint:  Check that the holdout set does not include genes in set A
+assert(data_holdout.shape[1] == (data.shape[1]-len(geneSetA)))
```

Comment:
This assertion won't actually check the specified checkpoint in line 167

maybe something like:

```python
assert(not(any(data_holdout.columns.isin(geneSetA))))
```

Reply:
This is more accurate.  Thank you for the suggestion